File: AllInOne/src/String/CountAndSay.java
package src.String;
/**
 * 
 * 38. Count and Say
 * 
 * @author jingjiejiang
 * @history Aor 27, 2022
 * 
 */
public class CountAndSay {
  public String countAndSay(int n) {

    assert n >= 1;
    String strRes = "1";

    for (int count = 2; count <= n ; count ++) {
      char preDigit = '0';
      int repeatCount = 0;
      StringBuilder strBuilder = new StringBuilder();

      for (int idx = 0; idx < strRes.length(); idx ++) {

        char curDigit = strRes.charAt(idx);

        if (curDigit != preDigit && idx != 0) {
          strBuilder.append(repeatCount).append(preDigit);
          repeatCount = 0;
        }

        repeatCount ++;
        preDigit = curDigit;
      }

      strBuilder.append(repeatCount).append(preDigit);
      strRes = strBuilder.toString();
    }
    return strRes;
  }
}

File: AllInOne/src/String/CountAndSay.js
/**
 * 
 * 38. Count and Say
 * 
 * @history Oct 23, 2022
 * 
 */
/**
 * @param {number} n
 * @return {string}
 */
 var countAndSay = function(n) {
    
  let strRes = '1';

  for (let count = 2; count <= n; count ++) {
    let preDigit = '0';
    let repeatCount = 0;
    let tmpString = '';

    for (let idx = 0; idx < strRes.length; idx ++) {
      if (idx != 0 && strRes[idx] != preDigit) {
        tmpString += repeatCount + preDigit;
        repeatCount = 0;
      }

      repeatCount ++;
      preDigit = strRes[idx];
    }

    tmpString += repeatCount + preDigit;
    strRes = tmpString;
  }

  return strRes;
};